src/nets/e2e_asr_attctc_th.py

```diff
@@ -797,8 +797,8 @@ def forward(self, enc_hs_pad, enc_hs_len, dec_z, att_prev, scaling=2.0):
 
         # weighted sum over flames
         # utt x hdim
-        # NOTE use bmm instead of sum(*)
-        c = torch.sum(self.enc_h * w.view(batch, self.h_length, 1), dim=1)
+        # NOTE equivalent to c = torch.sum(self.enc_h * w.view(batch, self.h_length, 1), dim=1)
+        c = torch.matmul(w.unsqueeze(1), self.enc_h).squeeze(1)
```

Comment:
using matmul is faster? (just my curiosity)

Reply:
The WSJ in both implementation is running now. Let's see tomorrow.

Reply:
in current status, the speedup is not so huge :disappointed: 
- sum(*) :  1.1633 iters/sec
- matmul :  1.2135 iters/sec

accuracies are almost same

Reply:
:(